### ISSUE TITLE
Error with `generateStubs` goal with Micronaut

### ISSUE THREAD

**Issue by ilopmar** [groovy/GMavenPlus#189]
We've discovered an issue compiling a Micronaut application using Groovy and Maven when executing the `generateStubs` goal.

Given the following Groovy interface:

```groovy
@CompileStatic
@Client(BintrayConfiguration.BINTRAY_API_URL)
interface BintrayClient {

    @Get('/api/${bintray.apiversion}/repos/${bintray.organization}/${bintray.repository}/packages')
    Flowable<BintrayPackage> fetchPackages()
}
```

The generated java code in `target/generated-sources/groovy-stubs/main/example/micronaut/BintrayClient.java` is not correct.

The value of the annotation `@Get` in the generated code should be exactly as it is in the Groovy source code, but the generated code contains a dot instead of the dollar sign:

```java
@groovy.transform.CompileStatic() @io.micronaut.http.client.annotation.Client(value="https://bintray.com")
 public interface BintrayClient
 {
;
@io.micronaut.http.annotation.Get(value="/api/.{bintray.apiversion}/repos/.{bintray.organization}/.{bintray.repository}/packages")
io.reactivex.Flowable<example.micronaut.BintrayPackage> fetchPackages();
}
```

So instead of `@Get(value="/api/${bintray.apiversion}")`, it is `@Get(value="/api/.{bintray.apiversion}")`. 

That dot makes the Micronaut application fails later.

I've tried using double quotes with:

```groovy
@Get("/api/\${bintray.apiversion}/repos/\${bintray.organization}/\${bintray.repository}/packages")     
```

But it doesn't make any difference.

In case you need an application to reproduce the issue, please use the following one. Just unzip it and run `./mvnw test`. You will see the compilation error showed before (because of a Micronaut compile-time check) related to the dollar-dot change I mentioned.

[micronaut-http-client-maven-groovy.tar.gz](https://github.com/groovy/GMavenPlus/files/6062553/micronaut-http-client-maven-groovy.tar.gz)

**Comment by keeganwitt** [comment]
Thanks for the example I see what you're talking about.  I'll start digging into Groovy as to why this is happening.

**Comment by keeganwitt** [comment]
It looks like this was an intentional change
https://github.com/apache/groovy/commit/561fad780794a5ac6be81818d2f123496e6c9bad
https://issues.apache.org/jira/browse/GROOVY-6085

I'm thinking about what the appropriate fix would be.

**Comment by ilopmar** [comment]
Thanks for the quick reply!

**Comment by keeganwitt** [comment]
This will obviously require a new Groovy release to fix.  But in the mean time, you might be able to workaround the issue by using the unicode equivalent rather than the literal character.
```groovy
@Get('/api/\u0024{bintray.apiversion}/repos/\u0024{bintray.organization}/\u0024{bintray.repository}/packages')
```

This left the dollar in the stub, but I'm not sure if this causes any obstacles with how you're using this annotation (I think it shouldn't as it's equivalent as far as Java is concerned, but you'll want to test that).

**Comment by ilopmar** [comment]
Which Groovy version are you using? In my test application it still uses the dot in the stub, not the dollar.

**Comment by keeganwitt** [comment]
I'd tested with 3.0.7.

**Comment by keeganwitt** [comment]
Oh shoot, you know what? I was looking at the decompiled class, not the stub.

**Comment by jameskleeh** [comment]
The issue here is that this logic should only happen if the expression is a class expression https://github.com/apache/groovy/commit/561fad780794a5ac6be81818d2f123496e6c9bad#diff-fb5c7aed8d75294b5fbca45b00da5e20968d33704cb81720503ee3601df30fecR732

**Comment by jameskleeh** [comment]
@keeganwitt We will create an issue in the Groovy tracker

**Comment by keeganwitt** [comment]
> The issue here is that this logic should only happen if the expression is a class expression [apache/groovy@561fad7#diff-fb5c7aed8d75294b5fbca45b00da5e20968d33704cb81720503ee3601df30fecR732](https://github.com/apache/groovy/commit/561fad780794a5ac6be81818d2f123496e6c9bad#diff-fb5c7aed8d75294b5fbca45b00da5e20968d33704cb81720503ee3601df30fecR732)

Exactly. I'm trying to think through how you could limit when this happens. It'd be pretty easy I think to exclude your case, since it should be a string literal, but I'm also wondering about static variables that aren't class references. I'm trying to think through the edge cases that need covered (currently working on tests to TDD this fix in).

**Comment by keeganwitt** [comment]
> @keeganwitt We will create an issue in the Groovy tracker

OK. Lemme know the issue number and I'll try to put in a fix.

**Comment by jameskleeh** [comment]
@keeganwitt https://issues.apache.org/jira/browse/GROOVY-9962

**Comment by ilopmar** [comment]
Thanks!

**Comment by keeganwitt** [comment]
My pleasure. Thanks for the report!
